src/redux/saga/employer.ts

```diff
@@ -1,19 +1,53 @@
-import { put, all, takeLatest, select } from 'redux-saga/effects';
-import axios from 'axios';
+import { put, takeLatest, select } from 'redux-saga/effects'
+import axios from 'axios'
+import { AppState } from '../types'
 
-import { registerEmployerSuccess } from '../../redux/actions/employer';
+import {
+  registerEmployerSuccess,
+  registerEmployerFail,
+  loginEmployerSuccess,
+  loginEmployerFail,
+} from '../../redux/actions/employer'
 
-function* registerEmployerSaga(credentials: Credential) {
+const credential = (state: AppState) => state.employer.credential
+function* registerEmployerSaga() {
   try {
-    const req = yield axios.post('/employer', { credentials });
-    yield put(registerEmployerSuccess(req.data));
+    const credentialData = yield select(credential)
+    const req = yield axios.post('/employer', {
+      info: {},
+      credential: credentialData,
+    })
+    yield put(registerEmployerSuccess(req.data))
   } catch (error) {
-    console.log(error);
+    yield put(registerEmployerFail())
   }
 }
 
+function* loginEmployerSaga() {
+  try {
+    const credentialData = yield select(credential)
+    const config = { headers: { Content_Type: 'Application/json' } }
```

Comment:
I think we don't need this since by default it sends `json` already

Reply:
Ok